xla/service/gpu/transforms/priority_fusion.cc

```diff
@@ -352,6 +353,48 @@ class PriorityFusionQueue {
     fusion_info_cache_.Invalidate(instruction);
   }
 
+  void UpdateRuntimes(
+      GpuPerformanceModel::RunTimes& runtimes, const HloInstruction* consumer,
+      const absl::flat_hash_map<const HloInstruction*, absl::Duration>&
+          original_consumers) {
+    auto it = original_consumers.find(consumer);
+    if (it != original_consumers.end()) {
+      runtimes.time_fused += it->second;
+      auto consumer_cache_result = gpu_performance_model_cache_.Get(*consumer);
+      CHECK(consumer_cache_result.has_value());
+      runtimes.time_unfused += (*consumer_cache_result).exec_time;
+    }
+  }
+
+  // Prepare for incremental updates
+  void ComputeRuntimesOfRemovedConsumers() {
+    for (auto pair : operands_to_new_consumers_) {
+      auto operand = pair.first;
+      auto reverse_it = reverse_map_.find(operand);
+      if (reverse_it == reverse_map_.end()) {
+        continue;
+      }
+      // this operand has been calculated priority before, we can do
+      // incremental update
+      // get all of this operand's original consumers
+      if (!gpu_performance_model_cache_.Contains(*operand)) {
+        // bitcast/constant have priority calculated but not in the cache
+        continue;
+      }
+      auto original_consumers =
```

Comment:
Can you please change this to `const auto &`?  Otherwise I suspect it's copying the `flat_hash_map`.  If a copy is intentional please leave a comment saying so.